server/utils/forms/inputs/durationInput.ts

```diff
@@ -0,0 +1,100 @@
+import * as ExpressValidator from 'express-validator'
+import { Request } from 'express'
+import Duration from '../../duration'
+import { FormValidationError } from '../../formValidationError'
+import { FormValidationResult } from '../formValidationResult'
+import FormUtils from '../../formUtils'
+
+export interface DurationErrorMessages {
+  empty: string
+  invalidDuration: string
+}
+
+export default class DurationInput {
+  constructor(
+    private readonly request: Request,
+    private readonly key: string,
+    private readonly errorMessages: DurationErrorMessages
+  ) {}
+
+  private get keys() {
+    return { hours: `${this.key}-hours`, minutes: `${this.key}-minutes` }
+  }
+
+  async validate(): Promise<FormValidationResult<Duration>> {
+    const result = await FormUtils.runValidations({ request: this.request, validations: this.validations })
+
+    const error = this.error(result)
+    if (error) {
+      return { value: null, error }
+    }
+
+    return {
+      value: this.duration!,
+      error: null,
+    }
+  }
+
+  private error(result: ExpressValidator.Result): FormValidationError | null {
+    const error = FormUtils.validationErrorFromResult(result)
```

Comment:
I'm finding the re-use of the word/variable `error` here a bit difficult to parse - I think it could boil down to us referring to `error.errors` in a few places - is there a more user-friendly variable name for what's returned from `validationErrorFormResult` that encapsulates the fact that it could contain multiple errors?

Reply:
Very good point - I've renamed it the variable to `formValidationError` to match the type.